presto-native-execution/presto_cpp/main/tests/TaskManagerTest.cpp

```diff
@@ -1004,11 +1004,37 @@ TEST_F(TaskManagerTest, distributedSort) {
   protocol::TaskUpdateRequest updateRequest;
   updateRequest.sources.push_back(
       makeRemoteSource("0", partialSortUris, 0, splitSequenceId));
-  std::string finalTaskId = taskIdGenerator.makeTaskId(0, 0);
-  auto finalSortTask =
-      createOrUpdateTask(finalTaskId, updateRequest, finalSortPlan);
-
+  const std::string finalTaskId = taskIdGenerator.makeTaskId(0, 0);
+  createOrUpdateTask(finalTaskId, updateRequest, finalSortPlan);
   assertResults(finalTaskId, rowType_, "SELECT * FROM tmp ORDER BY c0");
+
+  auto infoRequestState = http::CallbackRequestHandlerState::create();
+  auto statusRequestState = http::CallbackRequestHandlerState::create();
+  auto resultRequestState = http::CallbackRequestHandlerState::create();
+
+  const auto finalPrestoTaskStats =
+      taskManager_
+          ->getTaskInfo(
+              finalTaskId, false, std::nullopt, std::nullopt, infoRequestState)
+          .get()
+          ->stats;
+  const auto finalPrestoTaskStatus =
+      taskManager_
+          ->getTaskStatus(
+              finalTaskId, std::nullopt, std::nullopt, statusRequestState)
+          .wait()
+          .get();
+  ASSERT_EQ(
+      finalPrestoTaskStatus->peakNodeTotalMemoryReservationInBytes,
+      finalPrestoTaskStats.peakNodeTotalMemoryInBytes);
+  // Since we have more than one tasks in a query, then the task peak memory
```

Comment:
The comment suggests that the assert should be <=, not ==.

Reply:
Yeah, the test failed. Thanks and updated.